pkg/peer/subscription/audio.go

```diff
@@ -0,0 +1,60 @@
+package subscription
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/matrix-org/waterfall/pkg/common"
+	"github.com/pion/rtp"
+	"github.com/pion/webrtc/v3"
+)
+
+type AudioSubscription struct {
+	sender     *webrtc.RTPSender
+	controller SubscriptionController
+}
+
+func NewAudioSubscription(
+	outputTrack *webrtc.TrackLocalStaticRTP,
+	controller SubscriptionController,
+) (*AudioSubscription, error) {
+	if outputTrack == nil {
+		return nil, fmt.Errorf("Output track is nil")
+	}
+
+	sender, err := controller.AddTrack(outputTrack)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to add track: %s", err)
+	}
+
+	subscription := &AudioSubscription{sender, controller}
+	go subscription.readRTCP()
+
+	return subscription, nil
+}
+
+func (s *AudioSubscription) Unsubscribe() error {
+	return s.controller.RemoveTrack(s.sender)
+}
+
+func (s *AudioSubscription) WriteRTP(packet rtp.Packet) error {
+	return fmt.Errorf("Bug: no write RTP logic for an audio subscription!")
+}
+
+func (s *AudioSubscription) SwitchLayer(simulcast common.SimulcastLayer) {
+}
+
+func (s *AudioSubscription) Simulcast() common.SimulcastLayer {
+	return common.SimulcastLayerNone
+}
+
+func (s *AudioSubscription) readRTCP() {
+	for {
+		if _, _, err := s.sender.ReadRTCP(); err != nil {
+			if errors.Is(err, io.ErrClosedPipe) || errors.Is(err, io.EOF) {
+				return
+			}
+		}
+	}
```

Comment:
This seems like wasting performence?

Reply:
Hm? What do you mean?

Reply:
The for loop doesn't seem to be doing anything - why is there?

Reply:
To read RTCPs. I don't know if it's required for audio, but it seems like it's generally something that should be there since reading RTCPs (even if we don't process them) makes Pion process the packet with the interceptors, which is required for NACKs, etc.

Or did you mean something different?

Reply:
Ah, so `ReadRTCP()` does something even though we don't use the packets it returns?

Reply:
> Ah, so ReadRTCP() does something even though we don't use the packets it returns?

Yeah, right. Pion has a concept of interceptors and uses "default interceptors" if none others are specified. This construction is very [common in Pion examples](https://github.com/pion/webrtc/blob/master/examples/reflect/main.go#L77).

> (are there any RTCP packets for audio tracks even?)

I'm not an expert to be honest, but I assume that things like NACKs (negative acknowledgments) for audio do still exist (i.e. so that the sender indicates packet loss), since we still have sequence numbers (and timestamps) for audio packets (both video and audio are transmitted as RTP streams).

Reply:
Argh, right, this feels weird tbh, Pion. Can we add a comment?

Reply:
Added!